subsys/bootloader/debug/CMakeLists.txt

```diff
@@ -7,14 +7,17 @@
 set(EXTS ${ZEPHYR_BASE}/ext)
 set(SEGGER ${EXTS}/debug/segger)
 
-add_library(debuglib STATIC printk.c)
-target_link_libraries(debuglib PRIVATE zephyr_interface)
-
 add_subdirectory_ifdef(CONFIG_SB_DEBUG_PORT_UART uart)
-target_link_libraries_ifdef(CONFIG_SB_DEBUG_PORT_UART debuglib PRIVATE uart)
-
 add_subdirectory_ifdef(CONFIG_SB_DEBUG_PORT_SEGGER_RTT rtt)
-target_link_libraries_ifdef(CONFIG_SB_DEBUG_PORT_SEGGER_RTT debuglib PRIVATE segger_rtt)
 
-target_include_directories_ifdef(CONFIG_SB_DEBUG_PORT_SEGGER_RTT debuglib PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../include/debug)
-target_include_directories_ifdef(CONFIG_SB_DEBUG_PORT_SEGGER_RTT debuglib PUBLIC ${SEGGER}/rtt)
+zephyr_library()
+zephyr_library_sources(printk.c)
+
+zephyr_include_directories(
+  ../include/debug
+  )
+
+zephyr_include_directories_ifdef(
```

Comment:
Should this be in ./rtt/CMakeLists.txt instead?

Reply:
Fixed